src/esp/gfx/PbrShader.cpp

```diff
@@ -191,23 +189,25 @@ PbrShader::PbrShader(Flags originalFlags, unsigned int lightCount)
 
   // set texture binding points in the shader;
   // see PBR vertex, fragment shader code for details
-  if (flags_ & Flag::BaseColorTexture) {
+  if ((flags_ & Flag::BaseColorTexture) && lightCount_) {
     setUniform(uniformLocation("BaseColorTexture"), TextureUnit::BaseColor);
   }
-  if (flags_ & Flag::RoughnessTexture) {
+  if ((flags_ & Flag::RoughnessTexture) && lightCount_) {
```

Comment:
I'd maybe put all those ifs in a large `if(lightCount_)` block instead? So it's clearer what goes together. You're doing the same below for the defaults.